### ISSUE TITLE
False positive unused-variable for function-local field that begins with "unused_"

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [PyCQA/pylint#1345]
### Steps to reproduce
1. Run `pylint` on a file containing
```
def my_function():
    for index, (used_first_identifier, unused_second_identifier
               ) in enumerate(()):
        if index and used_first_identifier is None:
            break
```

### Current behavior
```
W:  2,39: Unused variable 'unused_second_identifier' (unused-variable)
```

### Expected behavior
The `unused-variable` warning will not be emitted because the identifier begins with the prefix "unused_".

### pylint --version output
```
__main__.py 1.6.5, 
astroid 1.4.9
Python 2.7.9 (default, Jun 29 2016, 13:08:31) 
[GCC 4.9.2]
```

**Comment by rogalski** [comment]
Standard Python idiom for throwaway name is `_`, not prepending original name with `unused_`.

If you want to allow all variables starting with `unused_` to be ignored, you need to change `dummy-variables-rgx` setting.

```
[VARIABLES]
# A regular expression matching the name of dummy variables (i.e. expectedly
# not used).
dummy-variables-rgx=_$|dummy|^unused_
```

**Comment by rogalski** [comment]
@nathanielmanistaatgoogle see source [in case you want to create a PR yourself]: https://github.com/PyCQA/pylint/blob/master/pylint/checkers/variables.py#L339

About changing a default - I'm not sure. Is Google style guide a popular style guide _outside of_ Google?

**Comment by nathanielmanistaatgoogle** [comment]
@rogalski: it's the designated Python style guide for all of Google's open source projects; I don't have good insight into how widely it's used by projects unaffiliated with Google.

**Comment by rogalski** [comment]
Let's ask @PCManticore

**Comment by PCManticore** [comment]
I think this could make sense as well, especially since it is a more readable alternative to ``_``, so @nathanielmanistaatgoogle feel free to submit a pull request with this.

**Comment by nathanielmanistaatgoogle** [comment]
@PCManticore, @rogalski: thanks; please take a look at [pull request 1357](https://github.com/PyCQA/pylint/pull/1357).
